src/cox_utils.jl

```diff
@@ -0,0 +1,78 @@
+# Compute first and last
+firsts(s) = [s[t].status && (t==1 || s[t] > s[t-1]) for t = 1:length(s)]
+lasts(s) = [s[t].status && (t==length(s) || s[t+1] > s[t]) for t = 1:length(s)]
+
+
+struct CoxSum{T,N}
+    values::Array{T,N}
+    sums::Array{T,N}
+    chunks::Array{T,N}
+    tails::Array{T,N}
+end
+
+function CoxSum(values::Array{T,N}, fs, ls) where {T, N}
+    sums = zeros(T, size(values,1)+1, Base.tail(size(values))...)
+    chunks = zeros(T, length(fs), Base.tail(size(values))...)
+    tails = zeros(T, length(fs), Base.tail(size(values))...)
+    CoxSum(values, sums, chunks, tails)
+end
+
+
+# fs = index first deaths, ls = index last deaths,
```

Comment:
Would be nice to use a more neutral term than "death". :-) Maybe "time-to-event" or just "time"?